src/datatypes.test.ts

```diff
@@ -224,6 +224,44 @@ describe("deserializeDatetime", () => {
     expect(deserializeDatetime("1990-11-12T10:00:00+14:00")).toEqual(
       expectedDateWithMaxPositiveTimezone
     );
+
+    const expectedEarliestRepresentableDate = new Date(-8640000000000000);
```

Comment:
I'd just make this (and the positive version) file-level constants with descriptive names - i.e. the same values, plus and minus, are used in a couple of different contexts (i.e. in Earliest, Latest, Earlier Timezone, in Later Timezone), e.g.
```
const EARLIEST_POSSIBLE_VALID_DATE = new Date(-8640000000000000);
const LATEST_POSSIBLE_VALID_DATE = new Date(8640000000000000);
```

Reply:
Huh, good point, those shouldn't have been different variables - the point of those tests is to represent the _same_ value with different serialisations. Fixed now.